fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java

```diff
@@ -63,6 +63,13 @@ public static double getRealCost(CostEstimate costEstimate) {
                 costEstimate.getNetworkCost() * networkCostWeight;
     }
 
+    public static int getParallelExecInstanceNum() {
+        if (ConnectContext.get().getSessionVariable().isEnablePipelineEngine()) {
+            return 1;
+        }
+        return Math.max(1, ConnectContext.get().getSessionVariable().getParallelExecInstanceNum());
```

Comment:
The real `ParallelExecInstanceNum` may be less than `getSessionVariable().getParallelExecInstanceNum()`, because  `LeftMostScanTabletsNum` may be very small. 
We should consider `getLeftMostScanTabletsNum` as before?

Reply:
For broadcast join, `getLeftMostScanTabletsNum ` has no affect with the cost,  we should consider the instance number instead?